File: Leetcode/SumofAllSubsetXORTotals.rs
// https://leetcode.com/problems/sum-of-all-subset-xor-totals

struct Solution {}

impl Solution {
    pub fn subset_xor_sum(nums: Vec<i32>) -> i32 {
        let n = nums.len();
        let mut ans = 0;

        for i in 1..(1 << n) {
            let mut x = 0;
            for j in 0..n {
                if i & (1 << j) > 0 {
                    x ^= nums[j];
                }
            }
            ans += x;
        }

        ans
    }
}

File: Leetcode/SumofAllSubsetXORTotals.c
// https://leetcode.com/problems/sum-of-all-subset-xor-totals

int subsetXORSum(int* nums, int numsSize) {
    int ans = 0;

    for (int mask = 0; mask < (1 << numsSize); ++mask) {
        int cur = 0;
        for (int i = 0; i < numsSize; ++i) {
            cur ^= ((mask & (1 << i)) > 0) * nums[i];
        }
        ans += cur;
    }

    return ans;
}